plugins/source/gcp/codegen/recipes/base.go

```diff
@@ -86,6 +105,11 @@ type Resource struct {
 	ExtraColumns []codegen.ColumnDefinition
 	// NameTransformer custom name transformer for resource
 	NameTransformer func(field reflect.StructField) (string, error)
+	// ClientOptions is a list of options to pass to the client
+	ClientOptions []string
+	// IgnoreInTestsColumns is a list of columns to ignore in tests
+	IgnoreInTestsColumns []string
+	ParentFilterFunc     string
```

Comment:
Apparently `ListLocations` returns invalid locations `us` and `eu` that can't be used to list sub resources, so we need to filter them. The function itself is manually written